src/components/Asset/AssetContent/CollectTokens.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { ReactElement, useState, useEffect } from 'react'
+import { useAsset } from '@context/Asset'
+import { useWeb3 } from '@context/Web3'
+import { gql, OperationResult } from 'urql'
+import { fetchData, getQueryContext } from '@utils/subgraph'
+import { FixedRateExchanges } from 'src/@types/subgraph/FixedRateExchanges'
+import Tooltip from '@shared/atoms/Tooltip'
+import { FixedRateExchange, LoggerInstance } from '@oceanprotocol/lib'
+import { getCollectTokensFeedback } from '@utils/feedback'
+import { getOceanConfig } from '@utils/ocean'
+import Loader from '@shared/atoms/Loader'
+
+const FixedRateExchangesQuery = gql`
+  query FixedRateExchanges($user: String, $exchangeId: String) {
+    fixedRateExchanges(where: { owner: $user, exchangeId: $exchangeId }) {
+      id
+      owner {
+        id
+      }
+      exchangeId
+      baseTokenBalance
+    }
+  }
+`
+
+export default function CollectTokens(): ReactElement {
+  const { asset, error, owner } = useAsset()
+  const [isOnwer, setIsOwner] = useState(false)
+  const [baseTokenBalance, setBaseTokenBalance] = useState(0)
+  const [isCollectLoading, setIsCollectLoading] = useState(false)
+  const [collectStatusText, setCollectStatusText] = useState('')
+
+  const { accountId, web3 } = useWeb3()
+
+  useEffect(() => {
+    if (!asset || error) {
+      return
+    }
+    setIsOwner(owner === accountId)
+  }, [asset, error])
+
+  useEffect(() => {
+    if (!accountId || asset.nft.owner !== accountId) return
```

Comment:
we should get the `owner` consistently throughout this file, but then in first effect we have this check done already  so should use here in the end:

```suggestion
    if (!accountId || !isOwner) return
```